core/input/input.cpp

```diff
@@ -1712,16 +1712,44 @@ void Input::add_joy_mapping(const String &p_mapping, bool p_update_existing) {
 }
 
 void Input::remove_joy_mapping(const String &p_guid) {
+	int fallback_mapping_offset = 0; // Fix the fallback, if we invalidate its index.
```

Comment:
```suggestion
	int index_removed = 0;
```
I think it might be simpler and more intuitive to use a variable to record the index position of the removal.

The old index can be used for comparison later. `index` greater than this value needs to be `index - 1`; `fallback_mapping` needs to be reset to `-1` if equal. Double loop can be avoided.

Reply:
I applied your feedback.

Please note that when adding mappings and when connecting joypads, the logic respects multiple mappings for the same GUID. It does not take the first match, but rather the last.  
With this change you requested, in the event that a guid with said multiple mappings is deleted, the index correction is not necessarily correct. The offset can be between 0 up to the amount of mappings which were deleted.

Reply:
There seems to be only one mapping for the same guid, but `fallback_mapping` can be shared by devices with other guids.

You can mark the case of removing `fallback_mapping`.  Then handle this situation in the following loop. The previous code handles this situation, so I think your previous code logic is not wrong, just not concise enough.

Reply:
Sorry, my mistake, a guid can have multiple mappings. Updating the index after the first mapping record of `p_guid` requires a second loop.

Reply:
No worries :)